url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/graph/EdgeView.java

```diff
@@ -0,0 +1,1787 @@
+								overlay(graph.getGraphics());
+							} else {
+								dirty.add(edge.getBounds());
+								graph.repaint((int) dirty.getX(), (int) dirty
+										.getY(), (int) dirty.getWidth(),
+										(int) dirty.getHeight());
+							}
+						}
+						return true;
+					} else if (!source
+							&& graph.getModel().acceptsTarget(cell, port)) {
+						if (edge.getTarget() != portView) {
+							edgeModified = true;
+							if (graph.isXorEnabled()) {
+								overlay(graph.getGraphics());
+							}
+							edge.setTarget(portView);
+							edge.update(graph.getGraphLayoutCache());
+							if (graph.isXorEnabled()) {
+								overlay(graph.getGraphics());
+							} else {
+								dirty.add(edge.getBounds());
+								graph.repaint((int) dirty.getX(), (int) dirty
+										.getY(), (int) dirty.getWidth(),
+										(int) dirty.getHeight());
+							}
+						}
+						return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		public boolean isConstrainedMoveEvent(MouseEvent e) {
+			GraphUI ui = graph.getUI();
+			if (ui instanceof BasicGraphUI)
+				return ((BasicGraphUI) ui).isConstrainedMoveEvent(e);
+			return false;
+		}
+
+		/**
+		 * Returning true signifies a mouse event adds a new point to an edge.
+		 */
+		public boolean isAddPointEvent(MouseEvent event) {
+			return event.isPopupTrigger()
+					|| SwingUtilities.isRightMouseButton(event);
+		}
+
+		/**
+		 * Returning true signifies a mouse event removes a given point.
+		 */
+		public boolean isRemovePointEvent(MouseEvent event) {
+			return event.isPopupTrigger()
+					|| SwingUtilities.isRightMouseButton(event);
+		}
+
+		protected boolean isSourceEditing() {
+			return source;
+		}
+
+		protected boolean isTargetEditing() {
+			return target;
+		}
+
+		/*
+		 * Returns true if either the source, target, label or a point is being
+		 * edited.
+		 */
+		protected boolean isEditing() {
+			return source || target || label || currentLabel >= 0
+					|| currentPoint != null;
+		}
+
+		/**
+		 * Invoked when the mouse pointer has been moved on a component (with no
+		 * buttons down).
+		 */
+		public void mouseMoved(MouseEvent event) {
+			for (int i = 0; i < r.length; i++)
+				if (r[i].contains(event.getPoint())) {
+					graph.setCursor(new Cursor(Cursor.CROSSHAIR_CURSOR));
+					event.consume();
+					return;
+				}
+			if (loc.contains(event.getPoint()) && graph.isMoveable()
+					&& GraphConstants.isMoveable(edge.getAllAttributes())) {
+				graph.setCursor(new Cursor(Cursor.HAND_CURSOR));
+				event.consume();
+			}
+			if (extraLabelLocations != null && graph.isMoveable()
+					&& GraphConstants.isMoveable(edge.getAllAttributes())) {
+				for (int i = 0; i < extraLabelLocations.length; i++) {
+					if (extraLabelLocations[i].contains(event.getPoint())) {
+						graph.setCursor(new Cursor(Cursor.HAND_CURSOR));
+						event.consume();
+					}
+				}
+			}
+		}
+
+		// Handle mouse pressed event.
+		public void mousePressed(MouseEvent event) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 39. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.